src/SFA.DAS.FindApprenticeship.Jobs/Application/Handlers/UpdateCandidateStatusHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using SFA.DAS.FindApprenticeship.Jobs.Domain;
+using SFA.DAS.FindApprenticeship.Jobs.Domain.Candidate;
+using SFA.DAS.FindApprenticeship.Jobs.Domain.Handlers;
+using SFA.DAS.FindApprenticeship.Jobs.Domain.Interfaces;
+
+namespace SFA.DAS.FindApprenticeship.Jobs.Application.Handlers
+{
+    public class UpdateCandidateStatusHandler(
+        ILogger<UpdateCandidateStatusHandler> logger,
+        IFindApprenticeshipJobsService findApprenticeshipJobsService,
+        IBatchTaskRunner batchTaskRunner) : IUpdateCandidateStatusHandler
+    {
+        public async Task BatchHandle(List<Candidate> candidates)
+        {
+            // Add tasks to the runner
+            for (var index = 1; index <= candidates.Count; index++)
+            {
+                var taskId = index;
+                var userReferenceIdIndex = index;
+                if (userReferenceIdIndex >= 0 && userReferenceIdIndex < candidates.Count)
```

Comment:
userReferenceIdIndex >= 0 will always be true here, as you initialise it with index which starts at 1. In the context of the function not sure why this var exists as it's just the value of index?

Reply:
Re-written the logic in better way